### ISSUE TITLE
states.service.running always succeeds when watched state has changes

### ISSUE THREAD

**Issue by fillarios** [saltstack/salt#43008]
### Description of Issue/Question

Applying `states.service.running` returns `True` even when systemd reports an error if the state it watches has changes.

Even running it with a nonexistent service name succeeds. 

### Setup

```yaml
trigger:
  cmd.run:
    - name: 'true'

service:
  service.running:
    - name: does_not_exist
    - watch:
      - trigger
```

Output:
```yaml
minion:
----------
          ID: trigger
    Function: cmd.run
        Name: true
      Result: True
     Comment: Command "true" run
     Started: 00:31:38.795226
    Duration: 21.271 ms
     Changes:   
              ----------
              pid:
                  23358
              retcode:
                  0
              stderr:
              stdout:
----------
          ID: service
    Function: service.running
        Name: does_not_exist
      Result: True
     Comment: Failed to start does_not_exist.service: Unit does_not_exist.service not found.
     Started: 00:31:39.845721
    Duration: 126.441 ms
     Changes:   

Summary for minion
------------
Succeeded: 2 (changed=1)
Failed:    0
------------
Total states run:     2
Total run time: 147.712 ms
```

### Versions Report

```yaml
Salt Version:
           Salt: 2017.7.0
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.5 (default, Nov  6 2016, 00:28:07)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4
 
System Versions:
           dist: centos 7.3.1611 Core
         locale: ANSI_X3.4-1968
        machine: x86_64
        release: 4.11.11-300.fc26.x86_64
         system: Linux
        version: CentOS Linux 7.3.1611 Core
```

**Comment by gtmanfred** [comment]
This will be fixed in PR #43016

Thanks for reporting.
Daniel

**Comment by fillarios** [comment]
Thank you! This PR fixes the issue (in my real use-case as well).
